gr-soapy/python/soapy/bindings/soapy_common.cc

```diff
@@ -49,22 +51,24 @@ static inline T string_to_setting(const std::string& str)
 }
 
 // Copied from SoapySDR 0.8
+//! convert empty and "false" strings to false, integers to their truthness
 template <>
 inline bool string_to_setting<bool>(const std::string& str)
 {
-    if (str == SOAPY_SDR_TRUE)
-        return true;
-    if (str == SOAPY_SDR_FALSE)
-        return false;
-
-    // zeros and empty strings are false
-    if (str == "0")
+    if (str.empty() || str == SOAPY_SDR_FALSE) {
         return false;
-    if (str == "0.0")
-        return false;
-    if (str == "")
-        return false;
-
+    }
+    if (str == SOAPY_SDR_TRUE) {
+        return true;
+    }
+    try {
+        return std::stoi(str);
```

Comment:
std::stoi("0.1") returns 0.

Reply:
oh yes ... let's just check for stod, as every integer should cleanly cast to double, anyway.

Reply:
Right.

In the qa on the SoapySDR PR,
`check_equal(SoapySDR::StringToSetting<bool>("0.2e12"), false);`
should be true
`check_equal(SoapySDR::StringToSetting<bool>("0.2"), true);`
should have failed the way it was coded

Reply:
yeah... shouldn't have written that code tired... fixed the unit test and both the conversion in this and the PR over at SoapySDR